catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -248,32 +253,136 @@ private void makeShallowCopiesDeepAgain(Project project) {
+				background.look.setZIndex(0);
+			}
 		}
-		if (currentlyEditedScene.getSpriteList().size() > 0) {
-			currentlyEditedScene.getSpriteList().get(0).setName(context.getString(R.string.background));
-			currentlyEditedScene.getSpriteList().get(0).look.setZIndex(0);
+	}
+
+	private static void initializeScripts(Project project) {
+		for (Scene scene : project.getSceneList()) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				for (Script script : sprite.getScriptList()) {
+					script.setParents();
+
+					if (script instanceof CollisionScript) {
+						((CollisionScript) script).updateSpriteToCollideWith(scene);
+					}
+				}
+			}
 		}
-		currentSprite = null;
+	}
 
-		PreferenceManager.getDefaultSharedPreferences(context)
-				.edit()
-				.putString(PREF_PROJECTNAME_KEY, project.getName())
-				.commit();
+	private static void loadLegoNXTSettingsFromProject(Project project, Context context) {
+		for (Setting setting : project.getSettings()) {
+			if (setting instanceof LegoNXTSetting) {
+				SettingsFragment.enableLegoMindstormsNXTBricks(context);
+				SettingsFragment.setLegoMindstormsNXTSensorMapping(context, ((LegoNXTSetting) setting).getSensorMapping());
+				return;
+			}
+		}
 	}
 
-	public void saveProject(Context context) {
-		if (project == null) {
-			return;
+	private static void loadLegoEV3SettingsFromProject(Project project, Context context) {
+		for (Setting setting : project.getSettings()) {
+			if (setting instanceof LegoEV3Setting) {
+				SettingsFragment.enableLegoMindstormsEV3Bricks(context);
+				SettingsFragment.setLegoMindstormsEV3SensorMapping(context, ((LegoEV3Setting) setting).getSensorMapping());
+				return;
+			}
 		}
+	}
 
-		project.saveLegoNXTSettingsToProject(context);
-		project.saveLegoEV3SettingsToProject(context);
+	@VisibleForTesting
+	public static void updateCollisionFormulasTo993(Project project) {
+		for (Scene scene : project.getSceneList()) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				for (Script script : sprite.getScriptList()) {
+					Brick scriptBrick = script.getScriptBrick();
+					if (scriptBrick instanceof FormulaBrick) {
+						FormulaBrick formulaBrick = (FormulaBrick) scriptBrick;
+						for (Formula formula : formulaBrick.getFormulas()) {
+							formula.updateCollisionFormulasToVersion();
+						}
+					}
+					for (Brick brick : script.getBrickList()) {
+						if (brick instanceof FormulaBrick) {
+							FormulaBrick formulaBrick = (FormulaBrick) brick;
+							for (Formula formula : formulaBrick.getFormulas()) {
+								formula.updateCollisionFormulasToVersion();
+							}
+						}
+					}
+				}
+			}
+		}
+	}
 
-		SaveProjectAsynchronousTask saveTask = new SaveProjectAsynchronousTask();
-		saveTask.execute();
+	private static void updateSetPenColorFormulasTo994(Project project) {
+		for (Scene scene : project.getSceneList()) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				for (Script script : sprite.getScriptList()) {
```

Comment:
what if we add a new method getAllScripts to reduce the intentation levels for the next few methods by 3?

Reply:
I just didn't add such a method because we would just need it for legacy conversion but I can do so if you insist?